### PR TITLE
[OGWA-7] As a user, I want to be able to change my password

### PR DESCRIPTION
**Pull request by nikkovanessa**

### CODE REVIEW

File: projects/client/src/Actions/user.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const loginAction = (data) => {
+    console.log("Data dari component ==>", data);
```
Comment (haikal1223) [inline projects/client/src/Actions/user.js]: remove console

File: projects/client/src/Reducers/user.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    // name: '',
+    // email: '',
+    // is_verified: 0,
+    // role_id: 0,
+    // profile_img: ''
+};
+
+export const userReducer = (state = INITIAL_STATE, action) => {
+    // action menerima 2 buah property ---> type & payload
+    console.log("Data dari component ==>", action);
```
Comment (haikal1223) [inline projects/client/src/Reducers/user.js]: remove console

File: projects/server/src/config/token.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const jwt = require('jsonwebtoken');
+
+module.exports = {
+    createToken: (payload, expired = '24h') => {
+        console.log(payload);
```
Comment (haikal1223) [inline projects/server/src/config/token.js]: remove console

File: projects/server/src/config/token.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    },
+    readToken: (req, res, next) => {
+        jwt.verify(req.token, 'xmart!', (err, decript) => {
+            if (err) {
+                return res.status(401).send({
+                    success: false,
+                    message: 'Authenticate token failed'
+                });
+            };
+            console.log(decript);
```
Comment (haikal1223) [inline projects/server/src/config/token.js]: remove console

File: projects/server/src/config/validator.js [hunk 5]
```diff
@@ -49,4 +49,86 @@ module.exports = {
...
+  validateForgot: async (req, res, next) => {
+    try {
+      await check("email")
+        .isString()
+        .notEmpty()
+        .isEmail()
+        .withMessage("Invalid email format")
+        .run(req)
+      const validation = validationResult(req);
+      console.log(validation);
```
Comment (haikal1223) [inline projects/server/src/config/validator.js]: remove console

File: projects/server/src/config/validator.js [hunk 6]
```diff
@@ -49,4 +49,86 @@ module.exports = {
...
+        .withMessage("Should contain at least 1 number")
+        .run(req);
+      await check("confirmpassword")
+        .isString()
+        .notEmpty()
+        .equals(req.body.password)
+        .withMessage("Password doesn't match")
+        .run(req);
+      const validation = validationResult(req);
+      console.log(validation);
```
Comment (haikal1223) [inline projects/server/src/config/validator.js]: remove console

File: projects/server/src/config/validator.js [hunk 7]
```diff
@@ -49,4 +49,86 @@ module.exports = {
...
+        .withMessage("Should more than 8 characters")
+        .matches(/[a-z]/g)
+        .withMessage("Should contain at least 1 lower case letter")
+        .matches(/[A-Z]/g)
+        .withMessage("Should contain at least 1 upper case letter")
+        .matches(/[0-9]/g)
+        .withMessage("Should contain at least 1 number")
+        .run(req);
+      const validation = validationResult(req);
+      console.log(validation);
```
Comment (haikal1223) [inline projects/server/src/config/validator.js]: remove console

**Review by gupitasekar — APPROVED** [review]

**Review by alvinnaldo — APPROVED** [review]

**Review by haikal1223 — APPROVED** [review]

**Pull request merged**